proposals/buildfile.md

```diff
@@ -0,0 +1,128 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```
+# "FROM" with option to use os/architecture for manifest lists
+baseImage: "gcr.io/distroless/java:8"
+baseImageOs: linux
+baseImageArchitecture: amd64
+
+creationTime: 0 # millis since epoch or iso8601 creation time
+format: Docker # Docker or OCI
+
+environment:
+  "MY_KEY1": "value"
+  "MY_KEY2": "value"
+labels:
+  "com.example.owner": "person"
+  "com.example.mode": "dev"
+volumes:
+  - "/myvolume"
+  - "/youvolume"
+
+exposedPorts:
+  - "123/udp"
+  - "456"
+  - "789/tcp"
+
+user: root
+workingDirectory: "/somewhere"
+entrypoint:
+  - "java"
+  - "-jar"
+cmd:
+  - "myjar.jar"
+
+# global file properties applied to all layers
+layerProperties:
+  filePermissions: "644"
+  directoryPermissions: "755"
+  user: "0"
+  group: "0"
+  timestamp: "0"
+layers:
+  - name: "scripts and classes"
+    # file properties only applied to this layer "scripts and classes"
+    properties:
+      filePermissions: "333"
+      directoryPermissions: "777"
+      user: "goose"
+      group: "3"
+      timestamp: "2020-06-03T19:31:50+00:00"
+    files:
+      - from: "target/scripts"
+        to: "/app/scripts"
+        # file properties only applied to this copy directive
+        properties:
+          filePermissions: "777"
+        # another copy for the same layer, with includes and excludes
+      - from: "target/classes"
+        to: "/app/classes"
+        excludes:
+          - "**/goose.class"
+          - "**/moose.class"
+        includes:
+          - "**/*.class"
+    # another layer, only globally defined file permissions are applied here
+  - name: "other"
+    files:
+      - from: "build/other"
+        to: "/app"
+    # a tar layer 
+  - name: "some tar layer"
```

Comment:
It'd be nice to support a foreign-layer and/or URL.  Ideally in such a way that it doesn't need to change to switch between `format: Docker` and `format: OCI`.

Reply:
> Ideally in such a way that it doesn't need to change to switch between format: Docker and format: OCI.

Assuming that is a goal.  I'm not sure it makes sense, especially if you're looking to explicitly support `mediaType`s.

Reply:
ah yes, forgot about this. Will add that in too.

Reply:
For a foreign layer (which must use a URL), I think `mediaType` already covers that?
```
"mediaType": "application/vnd.docker.image.rootfs.foreign.diff.tar.gzip"
```
And a tool could just support URL in the `archive` (or `archiveFile`) field if they want?

Reply:
maybe url is the right value?

Reply:
actually archive seems fine for now.